### PR TITLE
Incident Data Markers

### PR DESCRIPTION
**Pull request by JooRiiv**
ReportLayer now utilizes observations.tsx endpoint instead of mock data.

### CODE REVIEW

File: app/components/map/layers/ReportLayer.tsx [hunk 1]
```diff
@@ -1,54 +1,28 @@
...
-      longitude: 61.4973,
-      latitude: 23.7569,
-      description: "A towering oak tree standing for centuries.",
-    },
-  ];
+  const [reports, setReports] = useState<Observation[]>([]);
+
+  useEffect(() => {
+    async function fetchReports() {
+      const response = await fetch("/observations");
```
Comment (kuukkanen) [inline app/components/map/layers/ReportLayer.tsx:13]: Could you fetch these from a certain area? It's not very efficient to fetch them all at once.

File: app/routes/observations.tsx [hunk 2]
```diff
@@ -9,34 +9,39 @@ import { parseLatitude, parseLongitude } from "~/utils/parser";
...
-    startLatitude !== null &&
-    endLongitude !== null &&
-    endLatitude !== null
+    startLatitude === null ||
+    startLongitude === null ||
+    endLatitude === null ||
+    endLongitude === null
   ) {
-    const observations = await getObservations({
+    observations = await getObservations();
```
Comment (kuukkanen) [inline app/routes/observations.tsx:25]: It was intentional to leave the functionality of fetching all observations out of this endpoint. It's meant only for the admin to be able to fetch everything at once.

**Review by kuukkanen — CHANGES_REQUESTED** [review]

File: app/components/map/layers/ReportLayer.tsx (hunk 1 above)
Comment (JooRiiv) [reply app/components/map/layers/ReportLayer.tsx]: Done

File: app/routes/observations.tsx (hunk 2 above)
Comment (JooRiiv) [reply app/routes/observations.tsx]: Fixed

File: app/routes/observations.tsx [hunk 3]
```diff
@@ -56,13 +64,18 @@ export async function action({ request }: ActionFunctionArgs) {
...
+      !description ||
+      typeof description !== "string"
+    ) {
       return json({ error: "Missing required fields" }, { status: 400 });
     }
 
     let pictureKey: string | undefined;
 
-    if (picture) {
+    if (picture && picture instanceof File && picture.size > 0) {
```
[the remark this answers is not in the document]
Comment (kuukkanen) [reply app/routes/observations.tsx:78]: Do a rebase with main to fix things.

**Review by kuukkanen — CHANGES_REQUESTED** [review]
Try not to do merges in PRs.

**Review by kuukkanen — APPROVED** [review]
LGTM.

**Pull request merged**